rpc/src/rpc_filter.rs

```diff
@@ -0,0 +1,13 @@
+use {
+    solana_client::rpc_filter::RpcFilterType,
+    solana_sdk::account::{AccountSharedData, ReadableAccount},
+    spl_token_2022::{generic_token_account::GenericTokenAccount, state::Account},
+};
+
+pub fn satisfies_filter(account: &AccountSharedData, filter_type: &RpcFilterType) -> bool {
+    match filter_type {
+        RpcFilterType::DataSize(size) => account.data().len() as u64 == *size,
+        RpcFilterType::Memcmp(compare) => compare.bytes_match(account.data()),
+        RpcFilterType::TokenAccountState => Account::valid_account_data(account.data()),
+    }
+}
```

Comment:
nit, take it or leave it: since `RpcFilterType` already has a `verify` function, you could add this function to that `impl` and avoid this additional file

Reply:
Yeah, I actually did initially add a method to RpcFilterType that took an AccountSharedData.
The reason I ultimately went this route is that I didn't want to add a `solana-client` dependency on `spl-token-2022`. The alternative is adding a `solana-runtime` dependency and pulling in the inline-spl-token modules, but that didn't feel right either. In truth, both of those crates are sub-dependencies of client via `solana-transaction-status`, so probably I need to just get over myself 😝

Reply:
Ah gotcha, makes sense!